config/config.go

```diff
@@ -294,6 +294,16 @@ func (c *Config) Context() *Context {
 		return c.context
 	}
 	var hasher fosite.Hasher
+	var tracingEnabled bool
+
+	hasher = &fosite.BCrypt{
+		WorkFactor: c.BCryptWorkFactor,
+	}
+
+	if tracer, err := c.GetTracer(); err == nil && tracer.IsLoaded() {
+		tracingEnabled = true
+		hasher = &tracing.TracedBCrypt{c.BCryptWorkFactor}
```

Comment:
Please use keyed fields here

Reply:
I also noticed that this is kinda duplicated with [here](https://github.com/ory/hydra/blob/master/cmd/server/handler_oauth2_factory.go#L120-L122). I also noticed that [`hasher`](https://github.com/ory/hydra/blob/master/cmd/server/handler_oauth2_factory.go#L56) might be null if [tracing is disabled](https://github.com/ory/hydra/blob/master/cmd/server/handler_oauth2_factory.go#L120-L122). So instead this should use the bcrypt from the config (here)

Reply:
The nil thing is actually default behavior and did not cause any issues. `BCryptWorkFactor` was also set properly.

Reply:
I think for clarity it should we should use the hasher from the config [here](https://github.com/ory/hydra/blob/master/cmd/server/handler_oauth2_factory.go#L120-L122).

Reply:
> I also noticed that this is kinda duplicated with [here](https://github.com/ory/hydra/blob/master/cmd/server/handler_oauth2_factory.go#L120-L122). I also noticed that [`hasher`](https://github.com/ory/hydra/blob/master/cmd/server/handler_oauth2_factory.go#L56) might be null if [tracing is disabled](https://github.com/ory/hydra/blob/master/cmd/server/handler_oauth2_factory.go#L120-L122). So instead this should use the bcrypt from the config (here)

If Hasher is `nil` it gets set by fosite when Compose is called https://github.com/ory/fosite/blob/master/compose/compose.go#L56

Reply:
👍 on DRYing up the check to see if tracing has been loaded tho. I will clean that up.